src/Dynode/model/core/socket/inputSocket.ts

```diff
@@ -1,97 +1,92 @@
-import { JsonValue } from 'src/utils/objectUtils';
 import { VEventHandler, VEvent } from 'src/utils/vanillaEvent';
 import OutputSocket from './outputSocket';
 import Socket from './socket';
+import { Value } from './value';
 
 /**
  * Socket class that handles inputs.
  */
-class InputSocket extends Socket {
+class InputSocket<T> extends Socket<T> {
   /**
-   * Value used if the socket doesn't have a socket linked to it.
+   * Value used if the socket doesn't have a socket linked to it. null means there is nothing.
    */
-  private defaultValue: unknown;
-
-  /**
-   * If true, mark socket's default to be nothing.
-   */
-  private defaultNothing = false;
+  private defaultValue: Value<T> | null = null;
 
   /**
    * Marks whether socket can have a default value.
    */
-  private hasDefault = false;
+  private isDefaultSetVariable = false;
 
   /**
    * Reference to the socket that is connected to the input.
    */
-  linkedSocket: OutputSocket | undefined;
+  private linkedSocket: OutputSocket<T> | undefined;
 
   /**
    * Function that handles setting the value.
    */
-  valueHandler: VEventHandler | undefined;
+  private valueHandler: VEventHandler | undefined;
 
   /**
    * Provide a default value to socket.
    *
    * @param value  The new value. If omitted, defaultValue is set to nothing.
    */
-  setDefaultValue(value?: unknown): void {
-    if (arguments.length) {
-      this.defaultValue = value;
-      this.defaultNothing = false;
-    } else {
-      this.defaultNothing = true;
-    }
-    this.hasDefault = true;
+  public setDefaultValue(value: T): void {
+    this.defaultValue = new this.ValueType(value);
+    this.isDefaultSetVariable = true;
   }
 
   /**
    * Receive the default value that is set on the object.
    */
-  getDefaultValue(): unknown {
-    if (!this.hasDefault) throw Error('Input socket has no default value set');
-    if (this.defaultNothing) throw Error('Input socket default is nothing');
-    return this.defaultValue;
+  public getDefaultValue(): Value<T> {
+    // No need to check this.isDefaultSet because it's handled by this.isDefaultNothing
+    if (this.isDefaultNothing()) throw Error('Input socket default is nothing');
+    return this.defaultValue as Value<T>; // Can be cast because of "nothing" check
+  }
+
+  public setDefaultNothing(): void {
+    this.defaultValue = null;
+    this.isDefaultSetVariable = true;
   }
 
   /**
    * Check whether object's default value is set to nothing.
    */
-  isDefaultNothing() : boolean {
-    if (!this.hasDefault) throw Error('Input socket has no default value set');
-    return this.defaultNothing;
+  public isDefaultNothing(): boolean {
+    if (!this.isDefaultSet) throw Error('Input socket has no default value set');
```

Comment:
if default is not set, treat is as nothing state. So, just return true here, instead of throwing an exception.